### ISSUE TITLE
[apex] Add rule to report Apex LS semantic errors

### ISSUE THREAD

**Issue by KarlLivesey** [pmd/pmd#7002]
**Proposed Rule Name:** `ApexSemanticErrors`

**Proposed Category:** Error Prone

**Description:**

PMD's Apex multifile analysis already retains semantic issues produced by Apex LS, but those errors are not currently available as standard PMD rule violations.

`ApexSemanticErrors` would expose the existing Apex LS errors for each `ASTApexFile`. The rule would report only issues for which `Issue.isError()` is true, retain the original message and source range, and ignore warnings such as `Unused`. Multifile analysis would continue to use `PMD_APEX_ROOT_DIRECTORY`; the rule would not connect to an org or add a separate type checker.

The Apex LS issue API currently reports errors for incompatible assignments and return values, missing Apex types, invalid method arguments, and supported SObject field failures. Local custom-field metadata and cross-class references are resolved from the SFDX project.

One upstream limitation is worth documenting: after local metadata extends a standard object, Apex LS marks that object incomplete and does not report other unknown fields on it. The proposed rule should preserve that behavior instead of trying to fill the gap with a second semantic implementation.

**Code Sample:**

```apex
public class InvalidAssignments {
    private static void takesInteger(Integer value) {}

    public static Integer run() {
        Integer value = 'hello';
        takesInteger('hello');
        return 'hello';
    }
}
```

All three statements should be reported using the diagnostics already produced by Apex LS.

Valid local metadata must remain valid:

```apex
Account account = new Account();
account.Known_Text__c = 'hello';
```

when `Known_Text__c.field-meta.xml` is present in the SFDX project.

**Possible Properties:**

None. The rule is intended to be a direct `Apex LS Issue` to `PMD RuleViolation` adapter.
